addon/modifiers/enable-tooltip.ts

```diff
@@ -0,0 +1,217 @@
+  iconI.className = icon;
+  container.append(iconI);
+}
+
+function generateSubTitle(container: HTMLElement, subtitle?: string, html?: boolean): void {
+  if (!subtitle) return;
+  const subtitleSpan = document.createElement('span');
+  subtitleSpan.className = 'subtitle';
+  setElementContent(subtitleSpan, subtitle, html);
+  container.append(subtitleSpan);
+}
+
+function generateHTMLStructure(state: EnableTooltipState, args: EnableTooltipArgs): void {
+  const { title, subtitle, icon, html, placement } = args.named;
+  state.tooltip = document.createElement('div');
+  state.tooltip.className = 'upf-tooltip';
+  state.tooltip.dataset.placement = placement;
+  const titleContainer = document.createElement('div');
+  titleContainer.className = 'title-container';
+
+  generateIcon(titleContainer, icon);
+  generateTitle(titleContainer, title, html);
+  state.tooltip.append(titleContainer);
+  generateSubTitle(state.tooltip, subtitle, html);
+
+  if (isTesting()) {
+    document.querySelector('#ember-testing')?.append(state.tooltip);
+  } else {
+    document.body.append(state.tooltip);
+  }
+}
+
+function delayedRender(state: EnableTooltipState, args: EnableTooltipArgs): void {
+  if (state.isRendered) return;
+  state.setTimeoutId = setTimeout(() => {
+    renderTooltip(state, args);
+  }, RENDERING_DELAY);
+}
+
+function computePosition(state: EnableTooltipState, args: EnableTooltipArgs) {
+  const { placement } = args.named;
+  new Djoo().computePosition(
+    state.tooltip,
+    state.element,
+    {
+      placement: placement,
+      cssVariableName: 'modifier-tooltip',
+      targetOffset: 9,
+      viewportOffset: 16
+    },
+    { defaultRotation: 0, height: 4, width: 8 }
+  );
+}
+
+function renderTooltip(state: EnableTooltipState, args: EnableTooltipArgs): void {
+  if (state.isRendered) return;
+
+  generateHTMLStructure(state, args);
+  computePosition(state, args);
+
+  state.animation = createAnimation(state.tooltip, [{ opacity: 0 }, { opacity: 1 }], {
+    duration: ANIMATION_DURATION,
+    fill: 'forwards'
+  });
+  state.animation.play();
+  state.isRendered = true;
+  state.setTimeoutId = null;
+}
+
+function destroy(event: Event, state: EnableTooltipState): void {
+  if (state.setTimeoutId) {
+    clearTimeout(state.setTimeoutId);
+    state.setTimeoutId = null;
+    return;
+  }
+
+  const relatedTarget = (<MouseEvent>event).relatedTarget;
+  if (!state.isRendered || (relatedTarget instanceof Node && state.element.contains(relatedTarget))) return;
+
+  state.animation.reverse();
+  state.animation.finished.then(() => {
+    run(() => {
+      state.tooltip.remove();
+      state.isRendered = false;
+    });
+  });
+}
+
+function setDefaultConfiguration(args: EnableTooltipArgs): void {
+  args.named = {
+    ...args.named,
+    ...{
+      placement: args.named.placement || 'bottom',
+      trigger: args.named.trigger || 'hover focus',
+      title: args.named.title || '',
+      html: args.named.html || false
+    }
+  };
```

Comment:
not sure about mutating `args.named` directly like that instead of merging w/ defaults like:
```
const attrs = { ...args.named, ...DEFAULT_CONFIGURATION };
```

More globally, I noticed that `args.named` is passed around to a couple functions and I wonder if they should really know about the `named` part which is Modifier concept, or should just received an object directly